### PR TITLE
fix: Enable WinDSR with the new Azure CNI config

### PR DESCRIPTION
**Pull request by ShiqianTao**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
- The new Azure CNI config contains ```{windowsSettings:{}}```. Enable ```WinDSR``` under the ```windowsSettings```.
- Update UTs with the new Azure CNI config.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: staging/cse/windows/azurecnifunc.tests.suites/AzureCNI.Default.conflist [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
             "mode": "bridge",
             "bridge": "azure0",
+            "executionMode": "v4swift",
             "capabilities": {
                 "portMappings": true,
                 "dns": true
             },
             "ipam": {
-                "type": "azure-vnet-ipam"
+                "type": "azure-cns"
```
Comment (AbelHu) [inline staging/cse/windows/azurecnifunc.tests.suites/AzureCNI.Default.conflist:16]: Is this an expected change?

File: staging/cse/windows/azurecnifunc.tests.suites/AzureCNI.DisableWindowsOutboundNat.conflist [hunk 2]
```diff
@@ -7,12 +7,13 @@
...
             "mode": "bridge",
             "bridge": "azure0",
+            "executionMode": "v4swift",
             "capabilities": {
                 "portMappings": true,
                 "dns": true
             },
             "ipam": {
-                "type": "azure-vnet-ipam"
+                "type": "azure-cns"
```
Comment (jaer-tsun) [inline staging/cse/windows/azurecnifunc.tests.suites/AzureCNI.DisableWindowsOutboundNat.conflist:16]: I think this should still be azure-vnet-ipam?

File: staging/cse/windows/azurecnifunc.tests.suites/AzureCNI.Overlay.conflist [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
             "mode": "bridge",
             "bridge": "azure0",
+            "executionMode": "v4swift",
             "capabilities": {
                 "portMappings": true,
                 "dns": true
             },
             "ipam": {
-                "type": "azure-vnet-ipam"
+                "type": "azure-cns"
```
Comment (jaer-tsun) [inline staging/cse/windows/azurecnifunc.tests.suites/AzureCNI.Overlay.conflist:16]: not sure if it's necessary here but overlay also has [this](https://github.com/Azure/azure-container-networking/blob/95155ca9b4079cfe9ce4e93c04de419e28a3ad3c/cni/azure-windows-swift-overlay.conflist#L17) in config

File: staging/cse/windows/azurecnifunc.tests.ps1 [hunk 4]
```diff
@@ -89,17 +89,38 @@ Describe 'Set-AzureCNIConfig' {
...
     Context 'AzureCNIOverlay is enabled' {
-        It "Should not include Cluster CIDR when AzureCNIOverlay is enabled" {
+        It "Should not include Cluster CIDR when AzureCNIOverlay is enabled" {          
             Set-AzureCNIConfig -AzureCNIConfDir $azureCNIConfDir `
                 -KubeDnsSearchPath $kubeDnsSearchPath `
                 -KubeClusterCIDR $kubeClusterCIDR `
                 -KubeServiceCIDR $kubeServiceCIDR `
                 -VNetCIDR $vNetCIDR `
-                -IsDualStackEnabled $isDualStackEnabled `
-                -IsAzureCNIOverlayEnabled $true
```
Comment (AbelHu) [inline staging/cse/windows/azurecnifunc.tests.ps1:99]: Is this expected to remove `-IsAzureCNIOverlayEnabled $true`?

**Review by AbelHu — APPROVED** [review]

**Pull request merged**